File: app/src/main/java/com/ed/redditapp/ui/post_detail/PostDetailAdapter.java
package com.ed.redditapp.ui.post_detail;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.ed.redditapp.R;
import com.ed.redditapp.lib.api.Comment;

public class PostDetailAdapter extends RecyclerView.Adapter<PostDetailViewHolder> {
    private Comment[] comments;

    @NonNull
    @Override
    public PostDetailViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View itemView = LayoutInflater.from(parent.getContext()).inflate(R.layout.listitem_comment, parent, false);
        return new PostDetailViewHolder(itemView);
    }

    @Override
    public void onBindViewHolder(@NonNull PostDetailViewHolder holder, int position) {
        holder.setUsername(comments[position].getUsername());
        holder.setBody(comments[position].getBody());
        holder.setIndent(comments[position].getIndent());
    }

    @Override
    public int getItemCount() {
        if (comments == null) {
            return 0;
        }

        return comments.length;
    }

    public void updateComments(Comment[] comments) {
        this.comments = comments;
        notifyDataSetChanged();
    }
}

File: app/src/main/java/com/ed/redditapp/ui/post_detail/PostDetailAdapter.kt
package com.ed.redditapp.ui.post_detail

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.ed.redditapp.R
import com.ed.redditapp.lib.api.Comment

class PostDetailAdapter : RecyclerView.Adapter<PostDetailViewHolder>() {
    private var comments: Array<Comment> = emptyArray()

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): PostDetailViewHolder {
        val itemView = LayoutInflater
                .from(parent.context)
                .inflate(R.layout.listitem_comment, parent, false)
        return PostDetailViewHolder(itemView)
    }

    override fun onBindViewHolder(holder: PostDetailViewHolder, position: Int) {
        val comment = comments[position]
        holder.setInfo(comment.points, comment.timestamp, comment.username)
        holder.setBody(comment.body)
        holder.setIndent(comment.indent)
    }

    override fun getItemCount(): Int {
        return comments.size
    }

    fun updateComments(comments: Array<Comment>) {
        this.comments = comments
        notifyDataSetChanged()
    }
}